src/ORM/Table.php

```diff
@@ -1011,6 +1011,14 @@ public function findAll(Query $query, array $options)
      */
     public function findList(Query $query, array $options)
     {
+        if (!isset($options['keyField']) && !isset($options['valueField'])) {
+            $select = $query->clause('select');
+            if ($select && count($select) <= 2) {
+                $options['keyField'] = array_shift($select);
+                $options['valueField'] = array_shift($select) ?: $options['keyField'];
```

Comment:
I don't get your point.
Currently it does not work at all - this fixes it.
Current behavior using the documented definitions is not affected.
So it only makes things better.

Do you have a test case that shows what you mean?